modules/flyte/iam.tf

```diff
@@ -60,15 +60,29 @@ resource "aws_iam_role" "flyte_dataplane" {
           Federated = local.oidc_provider_arn
         }
         Condition : {
-          StringEquals : {
+          StringLike : {
```

Comment:
Oversight on my part in the initial PR, `StringEquals` doesn't work with the wildcard